

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Conversion Tutorial &mdash; SCelVis  documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Command Line Tutorial" href="tutorial_cli.html" />
    <link rel="prev" title="Analysis Tutorial" href="tutorial_analysis.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> SCelVis
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Tutorial</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="tutorial_analysis.html">Analysis Tutorial</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Conversion Tutorial</a></li>
<li class="toctree-l1"><a class="reference internal" href="tutorial_cli.html">Command Line Tutorial</a></li>
</ul>
<p class="caption"><span class="caption-text">Developer Info</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="developer_setup.html">Developer Setup</a></li>
<li class="toctree-l1"><a class="reference internal" href="developer_release.html">Releasing Packages</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">SCelVis</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Conversion Tutorial</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/tutorial_convert.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="conversion-tutorial">
<span id="tutorial-convert"></span><h1>Conversion Tutorial<a class="headerlink" href="#conversion-tutorial" title="Permalink to this headline">¶</a></h1>
<p>For visualization, data sets are provided as HDF5 files (<a class="reference external" href="https://anndata.readthedocs.io/en/latest/index.html">anndata</a> objects) that store gene expression (sparse CSR matrix) and meta data with very fast read access.
You can use <a class="reference external" href="http://scanpy.rtfd.io">scanpy</a> to create this HDF5 file directly or use the <code class="docutils literal notranslate"><span class="pre">scelvis</span> <span class="pre">convert</span></code> command for converting your single-cell pipeline output (see Section <a class="reference internal" href="tutorial_cli.html#tutorial-cli"><span class="std std-ref">Command Line Tutorial</span></a>).</p>
<div class="section" id="web-file-converter">
<h2>Web File Converter<a class="headerlink" href="#web-file-converter" title="Permalink to this headline">¶</a></h2>
<p>You can use the <span class="guilabel">Go To –&gt; Upload Data</span> menu entry to access the file conversion screen.</p>
<div class="figure align-center" id="id1">
<a class="reference internal image-reference" href="_images/scelvis_goto_upload.png"><img alt="_images/scelvis_goto_upload.png" src="_images/scelvis_goto_upload.png" style="width: 80%;" /></a>
<p class="caption"><span class="caption-text">Accessing the conversion screen.</span><a class="headerlink" href="#id1" title="Permalink to this image">¶</a></p>
</div>
<p>You can download an example ZIP archive with text format input from <a class="reference external" href="https://github.com/bihealth/scelvis/raw/master/examples/dummy_raw.zip">here</a>.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>&#64;Benedikt can you describe how this works? How would it work other formats.</p>
</div>
<p>You can find out about the structure of the source formats in the section <a class="reference internal" href="#tutorial-convert-formats"><span class="std std-ref">Source Format Details</span></a> below.</p>
</div>
<div class="section" id="source-format-details">
<span id="tutorial-convert-formats"></span><h2>Source Format Details<a class="headerlink" href="#source-format-details" title="Permalink to this headline">¶</a></h2>
<div class="section" id="hdf5-input">
<h3>HDF5 Input<a class="headerlink" href="#hdf5-input" title="Permalink to this headline">¶</a></h3>
<p>For HDF5 input, you can do your analysis with <a class="reference external" href="http://scanpy.rtfd.io">scanpy</a> to create an anndata object <code class="docutils literal notranslate"><span class="pre">ad</span></code>. SCelVis will use embedding coordinates from <code class="docutils literal notranslate"><span class="pre">ad.obsm</span></code>, cell annotation from <code class="docutils literal notranslate"><span class="pre">ad.obs</span></code> and expression data directly from <code class="docutils literal notranslate"><span class="pre">ad.X</span></code> (this should contain normalized and log-transformed expression values for all genes).
If present, information about the dataset will be extracted from strings stored in <code class="docutils literal notranslate"><span class="pre">ad.uns['about_title']</span></code>, <code class="docutils literal notranslate"><span class="pre">ad.uns['about_short_title']</span></code> and <code class="docutils literal notranslate"><span class="pre">ad.uns['about_readme']</span></code> (assumed to be Markdown).
Information about marker genes will be taken either from the <code class="docutils literal notranslate"><span class="pre">rank_genes_groups</span></code> slot in <code class="docutils literal notranslate"><span class="pre">ad.uns</span></code> or from entries starting with <code class="docutils literal notranslate"><span class="pre">marker_</span></code> in <code class="docutils literal notranslate"><span class="pre">ad.uns</span></code>: entries called <code class="docutils literal notranslate"><span class="pre">marker_gene</span></code> (required!), <code class="docutils literal notranslate"><span class="pre">marker_cluster</span></code>, <code class="docutils literal notranslate"><span class="pre">marker_padj</span></code>, <code class="docutils literal notranslate"><span class="pre">marker_LFC</span></code> will create a table with the columns <code class="docutils literal notranslate"><span class="pre">gene</span></code>, <code class="docutils literal notranslate"><span class="pre">cluster</span></code>, <code class="docutils literal notranslate"><span class="pre">padj</span></code>, and <code class="docutils literal notranslate"><span class="pre">LFC</span></code>.</p>
<p>If you prepared your data with <code class="docutils literal notranslate"><span class="pre">Seurat</span></code> (v2), you can use <code class="docutils literal notranslate"><span class="pre">Convert(from</span> <span class="pre">=</span> <span class="pre">sobj,</span> <span class="pre">to</span> <span class="pre">=</span> <span class="pre">&quot;anndata&quot;,</span> <span class="pre">filename</span> <span class="pre">=</span> <span class="pre">&quot;data.h5ad&quot;)</span></code> to get an HDF5 file.</p>
</div>
<div class="section" id="text-input">
<h3>Text Input<a class="headerlink" href="#text-input" title="Permalink to this headline">¶</a></h3>
<p>For “raw” text input, you need to prepare at least three files in the input directory:</p>
<ul>
<li><p><code class="docutils literal notranslate"><span class="pre">expression.tsv.gz</span></code>, a tab-separated file with normalized expression values for each gene (rows) in each cell (columns), e.g., like this:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">.</span>       <span class="n">cell_1</span>   <span class="n">cell_2</span>   <span class="n">cell_3</span>  <span class="o">...</span>
<span class="n">gene_1</span>  <span class="mf">0.13</span>     <span class="mf">0.0</span>      <span class="mf">1.5</span>     <span class="o">...</span>
<span class="n">gene_2</span>  <span class="mf">0.0</span>      <span class="mf">3.1</span>      <span class="mf">0.3</span>     <span class="o">...</span>
<span class="n">gene_3</span>  <span class="mf">0.0</span>      <span class="mf">0.0</span>      <span class="mf">0.0</span>     <span class="o">...</span>
</pre></div>
</div>
</li>
<li><p><code class="docutils literal notranslate"><span class="pre">annotation.tsv</span></code>, a tab-separated file with annotations for each cell, e.g., like this:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">.</span>         <span class="n">cluster</span>     <span class="n">genotype</span>  <span class="o">...</span>
<span class="n">cell_1</span>    <span class="n">cluster_1</span>   <span class="n">WT</span>        <span class="o">...</span>
<span class="n">cell_2</span>    <span class="n">cluster_2</span>   <span class="n">KO</span>        <span class="o">...</span>
</pre></div>
</div>
</li>
<li><p><code class="docutils literal notranslate"><span class="pre">coords.tsv</span></code>, a tab-separated file with embedding coordinates for each cell, e.g., like this:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">.</span>         <span class="n">tSNE_1</span>   <span class="n">tSNE_2</span>   <span class="n">UMAP_1</span>  <span class="n">UMAP_2</span>  <span class="o">...</span>
<span class="n">cell_1</span>    <span class="mf">20.53</span>    <span class="o">-</span><span class="mf">10.05</span>   <span class="mf">3.9</span>     <span class="mf">2.4</span>     <span class="o">...</span>
<span class="n">cell_2</span>    <span class="o">-</span><span class="mf">5.34</span>    <span class="mf">13.94</span>    <span class="o">-</span><span class="mf">1.3</span>    <span class="mf">3.4</span>     <span class="o">...</span>
</pre></div>
</div>
</li>
<li><p><code class="docutils literal notranslate"><span class="pre">markers.tsv</span></code>, an optional tab-separated file with marker genes and <strong>it needs to have a column named ``gene``</strong>, e.g., like this:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">gene</span>    <span class="n">cluster</span>     <span class="n">log2FC</span>   <span class="n">adj_pval</span>   <span class="o">...</span>
<span class="n">gene_1</span>  <span class="n">cluster_1</span>   <span class="mf">3.4</span>      <span class="mf">1.5e-6</span>     <span class="o">...</span>
<span class="n">gene_2</span>  <span class="n">cluster_1</span>   <span class="mf">1.3</span>      <span class="mf">0.00004</span>    <span class="o">...</span>
<span class="n">gene_3</span>  <span class="n">cluster_2</span>   <span class="mf">2.1</span>      <span class="mf">5.3e-9</span>     <span class="o">...</span>
</pre></div>
</div>
</li>
<li><p>a markdown file (e.g., <code class="docutils literal notranslate"><span class="pre">text_input.md</span></code>) with information about this dataset:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">----</span>
<span class="n">title</span><span class="p">:</span> <span class="n">An</span> <span class="n">Optional</span> <span class="n">Long</span> <span class="n">Data</span> <span class="n">Set</span> <span class="n">Title</span>
<span class="n">short_title</span><span class="p">:</span> <span class="n">optional</span> <span class="n">short</span> <span class="n">title</span>
<span class="o">----</span>

<span class="n">A</span> <span class="n">verbose</span> <span class="n">description</span> <span class="n">of</span> <span class="n">the</span> <span class="n">data</span> <span class="ow">in</span> <span class="n">Markdown</span> <span class="nb">format</span><span class="o">.</span>
</pre></div>
</div>
</li>
</ul>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>$ scelvis convert --input-dir text_input --output data/text_input.h5ad --about-md text_input.md
</pre></div>
</div>
<p>in <code class="docutils literal notranslate"><span class="pre">examples/dummy_raw.zip</span></code> and <code class="docutils literal notranslate"><span class="pre">examples/dummy_about.md</span></code> we provide raw data for a simulated dummy dataset.</p>
</div>
<div class="section" id="loom-input">
<h3>Loom Input<a class="headerlink" href="#loom-input" title="Permalink to this headline">¶</a></h3>
<p>for <a class="reference external" href="http://loompy.org">loompy</a> or <a class="reference external" href="https://github.com/mojaveazure/loomR">loomR</a> input, you can convert your data like this:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>$ scelvis convert --i input.loom -m markers.tsv -a about.md -o loom_input.h5ad
</pre></div>
</div>
<p>if you prepared your data with <code class="docutils literal notranslate"><span class="pre">Seurat</span></code> (v3), you can use <code class="docutils literal notranslate"><span class="pre">as.loom(sobj,</span> <span class="pre">filename=&quot;output.loom&quot;)</span></code> to get a <code class="docutils literal notranslate"><span class="pre">.loom</span></code> file and then convert to <code class="docutils literal notranslate"><span class="pre">.h5ad</span></code> with the above command.</p>
</div>
<div class="section" id="cellranger-input">
<h3>CellRanger Input<a class="headerlink" href="#cellranger-input" title="Permalink to this headline">¶</a></h3>
<p>Alternatively, the output directory of <code class="docutils literal notranslate"><span class="pre">CellRanger</span></code> can be used. This is the directory called <code class="docutils literal notranslate"><span class="pre">outs</span></code> containing either a file called <code class="docutils literal notranslate"><span class="pre">filtered_gene_bc_matrices_h5.h5</span></code> (version 2) or a file called <code class="docutils literal notranslate"><span class="pre">filtered_feature_bc_matrix.h5</span></code> (version 3), and a folder <code class="docutils literal notranslate"><span class="pre">analysis</span></code> with clustering, embedding and differential expression results. This will not no any further processing except log-normalization. Additionally, a markdown file provides meta information about the dataset (see above)</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>$ mkdir -p data
$ cat <span class="s">&lt;&lt;EOF &gt; data/cellranger.md</span>
<span class="s">----</span>
<span class="s">title: My Project</span>
<span class="s">short_title: my_project</span>
<span class="s">----</span>

<span class="s">This is my project data.</span>
<span class="s">EOF</span>
$ scelvis convert --input-dir cellranger-out --output data/cellranger_input.h5ad --about-md cellranger.md
</pre></div>
</div>
<p>In <code class="docutils literal notranslate"><span class="pre">examples/hgmm_1k_raw</span></code> we provide <code class="docutils literal notranslate"><span class="pre">CellRanger</span></code> output for the 1k 1:1 human mouse mix.
Specifically, from the <code class="docutils literal notranslate"><span class="pre">outs</span></code> folder we selected</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">filtered_feature_bc_matrix.h5</span></code></p></li>
<li><p>tSNE and PCA projections from <code class="docutils literal notranslate"><span class="pre">analysis/tsne</span></code> and <code class="docutils literal notranslate"><span class="pre">analysis/pca</span></code></p></li>
<li><p>clustering from <code class="docutils literal notranslate"><span class="pre">analysis/clustering/graphclust</span></code> and</p></li>
<li><p>markers from <code class="docutils literal notranslate"><span class="pre">analysis/diffexp/graphclust</span></code></p></li>
</ul>
<p><code class="docutils literal notranslate"><span class="pre">examples/hgmm_1k_about.md</span></code> contains information about this dataset.</p>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="tutorial_cli.html" class="btn btn-neutral float-right" title="Command Line Tutorial" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="tutorial_analysis.html" class="btn btn-neutral float-left" title="Analysis Tutorial" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2019-2020, BIH Core Unit Bioinformatics

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>