

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Analysis Tutorial &mdash; SCelVis  documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Conversion Tutorial" href="tutorial_convert.html" />
    <link rel="prev" title="SCelVis" href="index.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> SCelVis
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Tutorial</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Analysis Tutorial</a></li>
<li class="toctree-l1"><a class="reference internal" href="tutorial_convert.html">Conversion Tutorial</a></li>
<li class="toctree-l1"><a class="reference internal" href="tutorial_cli.html">Command Line Tutorial</a></li>
</ul>
<p class="caption"><span class="caption-text">Developer Info</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="developer_setup.html">Developer Setup</a></li>
<li class="toctree-l1"><a class="reference internal" href="developer_release.html">Releasing Packages</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">SCelVis</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Analysis Tutorial</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/tutorial_analysis.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="analysis-tutorial">
<span id="tutorial-analysis"></span><h1>Analysis Tutorial<a class="headerlink" href="#analysis-tutorial" title="Permalink to this headline">¶</a></h1>
<p>This tutorial describes the basics of performing the analysis of data using SCelVis.
For this tutorial, we will use the public demo instance at <a class="reference external" href="https://scelvis-demo.bihealth.org">https://scelvis-demo.bihealth.org</a>.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Data to be visualized can either be uploaded into a SCelVis server or it can be defined when the SCelVis server starts.
When using a remote SCelVis server such as the public instance at <a class="reference external" href="https://scelvis-demo.bihealth.org">scelvis-demo.bihealth.org</a>, you will most likely upload your data as shown below.
However, the server can also be started with the path or URL to the data.
This way, computational staff can provide predefined datasets to non-computational staff.
See <a class="reference internal" href="tutorial_cli.html#tutorial-cli"><span class="std std-ref">Command Line Tutorial</span></a> for more information.</p>
</div>
<p>First, obtain the file <a class="reference external" href="https://github.com/bihealth/scelvis/raw/master/examples/hgmm_1k.h5ad">hgmm_1k.h5ad</a> and download it to your computer.
This file contains data for 1000 cells from a 1:1 Mixture of Fresh Frozen Human (HEK293T) and Mouse (NIH3T3) Cells (10X v3 chemistry).
For a larger dataset, you could also use <a class="reference external" href="https://files.figshare.com/18037739/pbmc.h5ad">pbmc.h5ad</a> which is a published dataset of ~14000 IFN-beta treated and control PBMCs from 8 donors (<a class="reference external" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE96583">GSE96583</a>; see <a class="reference external" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE96583">Kang et al.</a>).</p>
<div class="section" id="upload-data">
<h2>Upload Data<a class="headerlink" href="#upload-data" title="Permalink to this headline">¶</a></h2>
<p>Then, use the menu on the top right to access the data upload screen <span class="guilabel">Go To –&gt; Upload Data</span>.</p>
<div class="figure align-center" id="id2">
<a class="reference internal image-reference" href="_images/scelvis_goto_upload.png"><img alt="_images/scelvis_goto_upload.png" src="_images/scelvis_goto_upload.png" style="width: 80%;" /></a>
<p class="caption"><span class="caption-text">Accessing the upload screen.</span><a class="headerlink" href="#id2" title="Permalink to this image">¶</a></p>
</div>
<p>On the next screen click the <span class="guilabel">Drag and Drop or Click and Select files.</span> and select the <em>hgmm_1k.h5ad</em> file for upload.
The file upload will take a while and you will be redirected to the data analysis screen shown in the next section.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Alternatively, you can use the example data set available through the top-right menu as <span class="guilabel">Go To –&gt; Data Sets –&gt; PBMC</span>.</p>
</div>
</div>
<div class="section" id="cell-annotation-analysis">
<h2>Cell Annotation Analysis<a class="headerlink" href="#cell-annotation-analysis" title="Permalink to this headline">¶</a></h2>
<p>In the beginning of each analysis, the cell annotation screen from the figure below is shown.</p>
<div class="section" id="scatter-plot">
<h3>Scatter Plot<a class="headerlink" href="#scatter-plot" title="Permalink to this headline">¶</a></h3>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>&#64;Benedikt: here we would need some more pointers towards the biological interpretation?</p>
</div>
<div class="figure align-center" id="id3">
<a class="reference internal image-reference" href="_images/scelvis_cell_scatter.png"><img alt="_images/scelvis_cell_scatter.png" src="_images/scelvis_cell_scatter.png" style="width: 80%;" /></a>
<p class="caption"><span class="caption-text">The cell annotation scatter plot.</span><a class="headerlink" href="#id3" title="Permalink to this image">¶</a></p>
</div>
<ol class="arabic">
<li><p><strong>Analysis selection tab.</strong>
This control allows to switch between</p>
<blockquote>
<div><dl class="simple">
<dt>About</dt><dd><p>A textual description attached to the dataset.</p>
</dd>
<dt>Cell Annotation</dt><dd><p>The cell annotation analysis screen that you are looking at.</p>
</dd>
<dt>Gene Expression</dt><dd><p>The gene expression analsysis screen.</p>
</dd>
</dl>
</div></blockquote>
</li>
<li><p><strong>Plot type selection.</strong>
This allows tos elect the plot type for the cell-based analysis.
Note that changing the plot type will change the subsequent controls.</p></li>
<li><p><strong>Cell filter button.</strong>
Using this control, you can filter cells based on various properties.
These includes the cluster, species and other properties.</p>
<p>NB: The values available here depend on the configuration of the analysis software generating the HDF5 file.</p>
</li>
<li><p><strong>Axis and color control.</strong>
This allows you to select the dimensions to display along the horizontal and vertical axes as well as the colouring.</p></li>
<li><p><strong>Differential expression button.</strong>
This allows you to run a differential expression between two groups of cells.
This will be explained in detail below in Section <a class="reference internal" href="#tutorial-cell-differential-expression"><span class="std std-ref">Differential Expression Analysis</span></a>.</p></li>
<li><p><strong>The cell scatter plot.</strong>
This is a dynamic scatter plot.
When doing cell annotation analysis, each dot corresponds to one cell.
The cells are colored by cluster by default but you can change this with the <span class="guilabel">select coloring</span> list.</p></li>
<li><p><strong>Plot controls.</strong>
When you move your mouse cursor over the plot then some buttons will appear on the top right.
These are explained in detail in Section <a class="reference internal" href="#tutorial-plot-ui-commands"><span class="std std-ref">Plot Interface Commands</span></a> together with some other tricks.</p></li>
<li><p><strong>Download data for this plot.</strong>
Download a CSV file with the data for reproducing the plot outside of SCelVis.</p></li>
</ol>
</div>
<div class="section" id="plot-interface-commands">
<span id="tutorial-plot-ui-commands"></span><h3>Plot Interface Commands<a class="headerlink" href="#plot-interface-commands" title="Permalink to this headline">¶</a></h3>
<p>The buttons on the top right of the plot are as follows.
NB: these are standard features of the <a class="reference external" href="https://plot.ly/">Plotly</a> library.</p>
<dl class="simple">
<dt>Save plot as image</dt><dd><p>The plot will be saved in PNG format.</p>
</dd>
<dt>Zoom</dt><dd><p>After clicking this button, you can select a rectangular area to zoom into.</p>
</dd>
<dt>Pan</dt><dd><p>Drag and drop the drawing area to move around in the plot.</p>
</dd>
<dt>Box Select, Lasso Select</dt><dd><p>After clicking this button, you can select a rectangular area on the plot or draw a free from shape to select dots.</p>
</dd>
<dt>Zoom In / Zoom Out</dt><dd><p>Zoom into or out of plot.</p>
</dd>
<dt>Autoscale / Reset Axes</dt><dd><p>This will reset the scaling to the original area.
You can obtain the same behaviour by double-clicking on a white spot in the plot.</p>
</dd>
<dt>Toggle Spike Lines</dt><dd><p>Enable horizontal and vertical lines when hovering over a point.</p>
</dd>
<dt>Show Closest / Compare Data on Hover</dt><dd><p>Change the spike lines behaviour.</p>
</dd>
</dl>
<p>Note that you can enable/disable individual groups by clicking their label in the legend.
You can disable all but one group by double-clicking the label.</p>
</div>
<div class="section" id="differential-expression-analysis">
<span id="tutorial-cell-differential-expression"></span><h3>Differential Expression Analysis<a class="headerlink" href="#differential-expression-analysis" title="Permalink to this headline">¶</a></h3>
<p>The differential expression analysis works as follows:</p>
<ol class="arabic simple">
<li><p>Click the <span class="guilabel">differential expression</span> button.
Four buttons appear: <span class="guilabel">group A</span>, <span class="guilabel">group B</span>, <span class="guilabel">reset</span>, <span class="guilabel">run</span>.</p></li>
<li><p>Click <span class="guilabel">group A</span>.
Then, use either the box or lasso select tool of the plot for selecting some points.
For example, click the lasso select button in the top of the right of the plot.
Move your mouse cursor to the position you want to start selecting at.
Keep the left mouse button pressed and draw a shape around the cells that you are interested in.
Release the mouse button.</p></li>
<li><p>Click <span class="guilabel">group B</span> and repeat step 2.</p></li>
<li><p>Click <span class="guilabel">run</span> to perform the analysis.</p></li>
</ol>
<p>The result could read something like <em>200 DE genes at 5% FDR</em>.
You can then click <span class="guilabel">view groups</span> to show the groups in the scatter plot, click <span class="guilabel">table</span> to see the resulting DE genes in the <span class="guilabel">Gene Expression</span> tab table.
You can also download the <span class="guilabel">results</span> or the <span class="guilabel">parameters</span> that were used for the DE gene analysis.</p>
<p>Clicking <span class="guilabel">reset</span> allows you to start a new DE gene analysis.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>&#64;Benedikt: here we would need a specific example with some possible interpretation.</p>
</div>
</div>
<div class="section" id="violin-plot">
<h3>Violin Plot<a class="headerlink" href="#violin-plot" title="Permalink to this headline">¶</a></h3>
<p>When selecting <span class="guilabel">violin plot</span> in the <span class="guilabel">select plot type</span> control you can draw violin plots.
For example, selecting <span class="guilabel">n_counts</span> and <span class="guilabel">nUMI_hg19</span> in the <span class="guilabel">select variable(s) and scaling</span> shows the following plot.
The top half shows the <em>nUMI_hg19</em> value distribution and the lower half shows the <em>n_counts</em> value distribution for the individual clusters.</p>
<div class="figure align-center" id="id4">
<a class="reference internal image-reference" href="_images/scelvis_cell_violin.png"><img alt="_images/scelvis_cell_violin.png" src="_images/scelvis_cell_violin.png" style="width: 80%;" /></a>
<p class="caption"><span class="caption-text">The cell annotation scatter plot.</span><a class="headerlink" href="#id4" title="Permalink to this image">¶</a></p>
</div>
<p>Use the <span class="guilabel">grouping</span> list to select the dimension to display along the horizontal axis.
You can use the <span class="guilabel">select split</span> list to select whether you want to further split the grouping.</p>
<p>Note that you can use a subset of the plot controls as described in Section <a class="reference internal" href="#tutorial-plot-ui-commands"><span class="std std-ref">Plot Interface Commands</span></a> and also download the data for the violin plot.
Hovering the mouse over the violin shapes will show you various statistical summaries of the distribution.</p>
</div>
<div class="section" id="bar-and-box-plots">
<h3>Bar and Box Plots<a class="headerlink" href="#bar-and-box-plots" title="Permalink to this headline">¶</a></h3>
<p>More visualization options are available by selecting the <span class="guilabel">bar plot</span> and <span class="guilabel">box plot</span> plot types.</p>
</div>
</div>
<div class="section" id="gene-expression-analysis">
<h2>Gene Expression Analysis<a class="headerlink" href="#gene-expression-analysis" title="Permalink to this headline">¶</a></h2>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>&#64;Benedikt: what about the show diffexp results?</p>
</div>
<p>When clicking the <span class="guilabel">Gene Expression</span> tab, you can perform a gene expression analysis.
As for the <span class="guilabel">Cell Annotation</span> analysis, it starts with the <span class="guilabel">scatter plot</span> type in (1).
However, first select some genes, for example <em>mm10_Hes1</em> and <em>hg19_HES1</em> in (2).</p>
<div class="section" id="id1">
<h3>Scatter Plot<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h3>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>&#64;Benedikt: here we would need some more pointers towards the biological interpretation?</p>
</div>
<div class="figure align-center" id="id5">
<a class="reference internal image-reference" href="_images/scelvis_gene_scatter.png"><img alt="_images/scelvis_gene_scatter.png" src="_images/scelvis_gene_scatter.png" style="width: 80%;" /></a>
<p class="caption"><span class="caption-text">The gene expression scatter plot.</span><a class="headerlink" href="#id5" title="Permalink to this image">¶</a></p>
</div>
<p>You can also display a table with the genes including their log2-fold change, counts and other information usin gthe <span class="guilabel">show marker table</span> button.
Using the check boxes on the left-hand-side (below; 1), you can select one or more genes and use the <span class="guilabel">use selected genes</span> (below; 2) button to add them to the <span class="guilabel">select gene(s)</span> field:</p>
<div class="figure align-center" id="id6">
<a class="reference internal image-reference" href="_images/scelvis_gene_table.png"><img alt="_images/scelvis_gene_table.png" src="_images/scelvis_gene_table.png" style="width: 80%;" /></a>
<p class="caption"><span class="caption-text">The gene expression scatter plot.</span><a class="headerlink" href="#id6" title="Permalink to this image">¶</a></p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>&#64;Benedikt: can you do some selection here and offer some biological interpretation?</p>
</div>
</div>
</div>
<div class="section" id="the-end">
<h2>The End<a class="headerlink" href="#the-end" title="Permalink to this headline">¶</a></h2>
<p>This is the end of the data analysis tutorial.
You might want to learn about the conversion of data into the HDF5 format next by reading Section <a class="reference internal" href="tutorial_convert.html#tutorial-convert"><span class="std std-ref">Conversion Tutorial</span></a>.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="tutorial_convert.html" class="btn btn-neutral float-right" title="Conversion Tutorial" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="index.html" class="btn btn-neutral float-left" title="SCelVis" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2019-2020, BIH Core Unit Bioinformatics

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>